pact-lang-api.js

```diff
@@ -292,19 +289,98 @@ var mkExp = function(pgmName) {
  * @return {object} of arguments, type-checked and properly named.
  */
 var mkMeta = function (sender, chainId, gasPrice, gasLimit) {
-  if (typeof sender !== 'string' ) {
-    throw new TypeError('sender must be a string: ' + JSON.stringify(sender));
-  }
-  if (typeof chainId !== 'string') {
-    throw new TypeError('chainId must be a string: ' + JSON.stringify(chainId));
-  }
-  if (typeof gasPrice !== 'number' ) {
-    throw new TypeError('gasPrice must be a number: ' + JSON.stringify(gasPrice));
-  }
-  if (typeof gasLimit !== 'number' ) {
-    throw new TypeError('gasLimit must be a number: ' + JSON.stringify(gasLimit));
+  enforceType(sender, "string", "sender");
+  enforceType(chainId, "string", "chainId");
+  enforceType(gasPrice, "number", "gasPrice");
+  enforceType(gasPrice, "number", "gasLimit");
+  return { "gasLimit":gasLimit, "chainId":chainId, "gasPrice":gasPrice, "sender":sender }
+}
+
+/**
+ * Formats ExecCmd into api request object
+ */
+var mkReq = function (cmd) {
+  return {
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    method: 'POST',
+    body: JSON.stringify(cmd),
   }
-  return {"gasLimit":gasLimit, "chainId":chainId, "gasPrice":gasPrice, "sender":sender}
+}
+
+/**
+ * Sends Pact command to a running Pact server and retrieves tx result.
+ * @param {Object}
```

Comment:
We need a `@typedef` for the command object documented here and `fetchLocal`, so we can simply say `@param [ref-to-command-typedef]`. There's some way to introduce a typedef that can be referenced elsewhere, I can't find it right now.